### PR TITLE
feat: logging

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

Global Logging solution with way of toggling it off and on as well as potential saving into file in future. 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/configs/bootstrap.dart [hunk 1]
```diff
@@ -70,8 +84,44 @@ Future<void> _runApp(Widget app) async {
...
+  if (kDebugMode) {
+    debugPrint('PlatformDispatcher Error');
+    debugPrint(error.toString());
+    debugPrintStack(stackTrace: stack, label: 'PlatformDispatcher');
+  }
+
+  return true;
+}
+
+/// Uncouth Errors reporting
```
Comment (dtscalac) [inline catalyst_voices/lib/configs/bootstrap.dart:120]:
```suggestion
/// Uncaught Errors reporting
```

File: catalyst_voices/lib/dependency/dependencies.dart [hunk 2]
```diff
@@ -46,4 +47,11 @@ final class Dependencies extends DependencyProvider {
       SecureStorageService(),
     );
   }
+
+  void _registerManagers() {
```
[the remark this answers is not in the document]
Comment (damian-molinski) [reply catalyst_voices/lib/dependency/dependencies.dart:51]: Well, none. Will change

File: catalyst_voices/lib/configs/bootstrap.dart [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:go_router/go_router.dart';
 import 'package:url_strategy/url_strategy.dart';
 
+final _loggingService = LoggingService();
+
+final _bootstrapLogger = Logger('Bootstrap');
+final _flutterLogger = Logger('Flutter');
+final _platformDispatcherLogger = Logger('PlatformDispatcher');
+final _uncouthZoneLogger = Logger('UncouthZone');
```
Comment (dtscalac) [inline catalyst_voices/lib/configs/bootstrap.dart:21]:
```suggestion
final _uncaughtZoneLogger = Logger('UncaughtZone');
```

File: catalyst_voices/lib/configs/bootstrap.dart [hunk 4]
```diff
@@ -23,26 +30,40 @@ final class BootstrapArgs {
...
 // TODO(damian-molinski): Add Isolate.current.addErrorListener
-// TODO(damian-molinski): Add runZonedGuarded
-// TODO(damian-molinski): Add Global try-catch
 Future<void> bootstrap([
-  BootstrapWidgetBuilder builder = _appBuilder,
+  BootstrapWidgetBuilder builder = _defaultBuilder,
 ]) async {
+  runZonedGuarded(
+    () => _safeBootstrap(builder),
+    _reportUncouthZoneError,
```
Comment (dtscalac) [inline catalyst_voices/lib/configs/bootstrap.dart:39]:
```suggestion
    _reportUncaughtZoneError,
```

File: catalyst_voices/lib/configs/bootstrap.dart [hunk 5]
```diff
@@ -70,8 +90,32 @@ Future<void> _runApp(Widget app) async {
...
+}
+
+/// Platform Dispatcher Errors reporting
+bool _reportPlatformDispatcherError(Object error, StackTrace stack) {
+  _platformDispatcherLogger.severe('Platform Error', error, stack);
+  return true;
+}
+
+/// Uncaught Errors reporting
+void _reportUncouthZoneError(Object error, StackTrace stack) {
```
Comment (dtscalac) [inline catalyst_voices/lib/configs/bootstrap.dart:119]:
```suggestion
void _reportUncaughtZoneError(Object error, StackTrace stack) {
```

File: catalyst_voices/lib/configs/bootstrap.dart [hunk 6]
```diff
@@ -70,8 +90,32 @@ Future<void> _runApp(Widget app) async {
...
+  _uncouthZoneLogger.severe('Uncouth Error', error, stack);
```
Comment (dtscalac) [inline catalyst_voices/lib/configs/bootstrap.dart:120]:
```suggestion
  _uncaughtZoneLogger.severe('Uncouth Error', error, stack);
```

**Review by dtscalac — APPROVED** [review]
lgtm

**Pull request merged**